### ISSUE TITLE
config cache problem

### ISSUE THREAD

**Issue by kudlohlavec** [sfelix-martins/passport-multiauth#62]
Hi,

I am encountering strange issue with multiauth. I am using Laravel inside of docker and right after startup of the docker container everything seems to work as it should, however after few minutes strange things start to happen 😃 .

When I am trying to login through original Laravel User model I got this error:

`Client error: POST localhost/oauth/token resulted in a 401 Unauthorized response:{\"error\":\"invalid_client\",\"message\":\"Client authentication failed\"}`

What is pretty self-describing, it cannot match passport client with request, however everything is being set properly, because right after startup of docker it worked. Solution for this problem is to run `php artisan config:clear` . After this command everything works fine so It looks like there is some problem with the config cache, I guess?

Another error related to this issue jumps when I am trying to login through my own added model called Gateway. The symptoms are similar to the issue above, after startup everything works fine and after few minutes this error occurs:

`Server error: POST localhost/oauth/token resulted in a 500 Internal Server Error response:<!DOCTYPE html><!--League\\OAuth2\\Server\\Exception\\OAuthServerException: The request is missing a required parameter,  (truncated...)"`

Again, it is clear to me what it is talking about - it is missing provider parameter from the request, however it is present and was working right after startup. Again, after running of `php artisan config:clear` everything seems to be working properly.

And my last observation regarding config cache issue is when after few minutes after startup I am trying to access protected endpoint by Bearer token retrieved right after startup. This is the error, please note that I shortened token stated in the error:

`SQLSTATE[42S22]: Column not found: 1054 Unknown column 'api_token' in 'where clause' (SQL: select * from users where api_token = eyJ0eXAiOiJKV1QiLCJhb limit 1`

Solution for this problem is again in command `php artisan config:clear`

All leads seem to be pointing into config cache related issue. Hard to say if it is caused by docker environment or not, however these are my tech specs:

Docker for Mac - 18.06.0-ce
Laravel - 5.7.13
Passport - 7.0.3
Multiauth - 4.0

**Comment by kudlohlavec** [comment]
Ok, so it looks like I have figured it out. Indeed, problem was in caching of config, however the real problem was in syncing service that Ive been using altogether with docker in order to "copy" code developed locally into docker container. Along with source files there were also copied config cache file (bootstrap/cache/config.php). This syncing is being performed continuously and takes some time so therefore the issue occurred always few minutes after startup and not immediately.

TL;DR - Docker users: Ensure that you are not copying bootstrap/cache/config.php file into docker container in order to avoid above mentioned problems.
